modules/parent-join/src/main/java/org/elasticsearch/join/query/HasParentQueryBuilder.java

```diff
@@ -74,7 +74,7 @@ public HasParentQueryBuilder(String type, QueryBuilder query, boolean score) {
     }
 
     private HasParentQueryBuilder(String type, QueryBuilder query, boolean score, InnerHitBuilder innerHitBuilder) {
-        this.type = requireValue(type, "[" + NAME + "] requires 'type' field");
+        this.type = requireValue(type, "[" + NAME + "] requires 'parent_type' field");
```

Comment:
To avoid this getting out-of-sync in the future, it would be nice to use `PARENT_TYPE_FIELD.getPreferredName()` instead of hard-coding `parent_type`. The same idea applies to `query` below.